doc/setup-binderhub.rst

```diff
@@ -1,104 +1,147 @@
-   address. This means that you need to have control over a URL such as
-   ``www.mydomain.com``.
-
-Download the Helm Chart and deployment repo for BinderHub
----------------------------------------------------------
-
 BinderHub uses Helm Charts to set up the applications we'll use in our Binder
 deployment. If you're curious about what Helm Charts are and how they're
 used here, see the `Zero to JupyterHub guide
 <https://zero-to-jupyterhub.readthedocs.io/en/latest/tools.html#helm>`_.
 
-First grab the latest helm chart for BinderHub and install it.::
-
-    helm repo add jupyterhub https://jupyterhub.github.io/helm-chart
-    helm install --name binder jupyterhub/binderhub --version v0.1.0-464c4f5
-
-
-Configure DNS
--------------
+Below we'll cover how to configure your Helm Chart, and how to create your
+BinderHub deployment.
 
-Next we'll configure our web address to point to the IP address for our
-BinderHub.
+Preparing to install
+--------------------
 
-Find the ``EXTERNAL-IP`` of the **nginx-ingress-controller**::
+To configure the Helm Chart we'll need to generate and insert a few pieces of
+information.
 
-    kubectl --namespace=support get svc support-nginx-ingress-controller
+The first is the content of the JSON file created when we set up
+the container registry. For more information on getting a registry password, see
+:ref:`setup-registry`. We'll copy/paste the contents of this file in the steps
+below.
 
-Make DNS records for both binder and jupyterhub that point to that IP. For
-example, make both::
+We also need two random tokens to configure out BinderHub. Generate these
+tokens by running the following commands then copying the outputs.::
 
-   hub.mydomain.com
-   binder.mydomain.com
-
-point to the same ``EXTERNAL-IP`` address of the ingress controller.
+    openssl rand -hex 32
+    openssl rand -hex 32
 
 .. note::
 
-   When you configure the helm chart, you will add the DNS entry for binder and
-   jupyterhub in the ``config.yaml`` file.
-
+   This command is run **twice** because we need two different tokens.
 
-Configure the Helm Chart
-------------------------
+Create ``secret.yaml``
+----------------------
 
-First create a file called ``secret.yaml``. In it, put the following code::
+Create a file called ``secret.yaml``. In it, put the following code::
 
-    jupyterhub:
+  jupyterhub:
       hub:
-        cookieSecret: "<openssl rand -hex 32>"
         services:
           binder:
-            apiToken: "<openssl rand -hex 32>"
+            apiToken: "<output of FIRST `openssl rand -hex 32`>"
       proxy:
-        secretToken: "<openssl rand -hex 32>"
-    registry:
-      password: |
-        <the json file from Service Accounts>
-    hub:
-      services:
-        binder:
-          apiToken: "<openssl rand -hex 32>"
+        secretToken: "<output of SECOND `openssl rand -hex 32`>"
+  registry:
+    password: |
+      <contents of the json file from Service Accounts>
```

Comment:
Service Accounts is a new concept here. <contents of the JSON file for the container registry password>